### ISSUE TITLE
Prevent recalculation algorithm if no yeild is given

### ISSUE THREAD

**Issue by christianlupus** [nextcloud/cookbook#1986]
The recalucaltion algorithm of the ingredients fails if the recipe yield is non-numeric or null. This will result in NaN entries while recipe viewing. This should be fixed.

**Comment by mohmaaya** [comment]
I'd like to fix this

**Comment by christianlupus** [comment]
I will assign you happily!

**Comment by christianlupus** [comment]
If you need help, feel free to contact us here, in the discussions or the matrix chat.

**Comment by mohmaaya** [comment]
Thanks, I ll work on it in sometime. :)

**Comment by mohmaaya** [comment]
Hi Christian,
I need a little help to understand how does the app work to run it locally.
Does the nextcloud/server acts as the backend? Or the cookbook is self contained because I am getting "Cannot /GET" at the moment when running locally.
Thanks

**Comment by christianlupus** [comment]
Hello @mohmaaya. There is the documentation page at https://nextcloud.github.io/cookbook/dev/contributing/setup that suggests mainly Julius' development setup. Did you find that and follow its process?

**Comment by stefan123t** [comment]
I do not know what caused the NaN in your case and therefor whether the fix from @j0hannesr0th will fix my newbie issue too ?

I have found that I got NaN for all ingredients until I entered the number of portions into the servings entry field.
Because it did not assume `1` as the default serving it was probably calculating / multiplying by 0 and that was reported as NaN.

Cheers for this fine piece of software!

I only had some minor difficulties as it is not yet readily available from the nextcloud app store and the [documentation for administrators](https://nextcloud.github.io/cookbook/admin/) is still missing:

* Download the correct version (Stable channel) from https://apps.nextcloud.com/apps/cookbook
In my case I had to download [0.10.5](https://github.com/christianlupus-nextcloud/cookbook-releases/releases/download/v0.10.5/cookbook-0.10.5.tar.gz) for Nextcloud 27.

* Install it into your `/nextcloud/apps` folder
  ```
  cd /nextcloud/apps
  wget https://github.com/christianlupus-nextcloud/cookbook-releases/releases/download/v0.10.5/cookbook-0.10.5.tar.gz
  tar -xvzf cookbook-0.10.5.tar.gz
  ```

* Enable Cookbook App under Profile > + Apps > Disabled Apps
  Cookbook (Enable)

**Comment by christianlupus** [comment]
@mohmaaya may I ask you if you had any success so far? It it progressing?

**Comment by mohmaaya** [comment]
@christianlupus I am so sorry but I got very busy that I completely forgot about this task. I can work on this in the next couple of days since I have time. Can you please let me know if the bug is fixed with johannes PR as I see above merge. If so, then I can take another task.

**Comment by christianlupus** [comment]
No problem. I just wanted to make sure we are not waiting on each other for some reason.

The issue is not solved yet. The provided PR was addressing some other issue. The actual problem in this issue is the division by zero in case the recipe yield cannot be parsed successfully. This is still the case.

**Comment by MikeSpaceG** [comment]
Is there a way to disable this "recalculation"? Most of my recipes now display recalc errors or NaN. I cannot find any other reference to this "feature."

**Comment by christianlupus** [comment]
I think you can create a recipe simply. Just create a new recipe, give it a random name and put some ingredients with a number (so that the calculation is triggered). Make sure to _not_ set the yield field (or set it to null). You will most probably see the NaNs due to x/0 calculation.

I have not tested right now the steps as I am on mobile but I tried it multiple times.

**Comment by christianlupus** [comment]
Here is a short video on the issue:

https://github.com/nextcloud/cookbook/assets/8202922/91e5e89c-d406-46dc-b963-22851892d04f

**Comment by christianlupus** [comment]
@mohmaaya wanted to work on this issue, @j0hannesr0th. Please sync to avoid unnecessary doubled work and frustration on either side. Thanks.
